utils/firestore.js

```diff
@@ -3013,13 +3013,12 @@ const getSpecimensByCollectionIds = async (collectionIdsArray, siteCode, isBPTL
 }
 
 const processSendGridEvent = async (event) => {
-    if (event.gcloud_project !== process.env.GCLOUD_PROJECT) return;
+    if (!event.notification_id || event.gcloud_project !== process.env.GCLOUD_PROJECT) return;
```

Comment:
I'm curious. Do we have mixed events from different tiers (dev, stage, prod)?

Reply:
You are correct. so i used  event.gcloud_project to detect the right tier.